src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -1637,17 +1669,41 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     // Now if all of the tasks are blocked, then we need to break a deadlock
     bool ret = all_task_ids.size() == blocked_task_ids.size() && !all_task_ids.empty();
     if (ret) {
-      logger->info(
-        "deadlock state is reached with all_task_ids size: {}, blocked_task_ids: {}, "
-        "bufn_task_ids: {}, threads size: {}",
-        all_task_ids.size(),
-        blocked_task_ids.size(),
-        bufn_task_ids.size(),
-        threads.size());
+      std::set<long> threadsKeySet;
+      std::transform(threads.begin(),
+                     threads.end(),
+                     std::inserter(threadsKeySet, threadsKeySet.begin()),
+                     [](const auto& pair) { return pair.first; });
```

Comment:
```suggestion
      for(auto const& [k, v] : threads) { thread_key_set.insert(k); }
```

Reply:
In addition, this `std::set` is unused if `is_log_enabled` is `false`, so it should better be declared inside that `if` branch.